File: app/src/main/java/com/ryanrvldo/moviecatalogue/utils/EspressoIdlingResource.java
package com.ryanrvldo.moviecatalogue.utils;

import androidx.test.espresso.IdlingResource;
import androidx.test.espresso.idling.CountingIdlingResource;

public class EspressoIdlingResource {

    private static final String RESOURCE = "GLOBAL";
    private static CountingIdlingResource idlingResource = new CountingIdlingResource(RESOURCE);

    public static void increment() {
        idlingResource.increment();
    }

    public static void decrement() {
        idlingResource.decrement();
    }

    public static IdlingResource getIdlingResource() {
        return idlingResource;
    }
}

File: app/src/main/java/com/ryanrvldo/moviecatalogue/utils/EspressoIdlingResource.kt
package com.ryanrvldo.moviecatalogue.utils

import androidx.test.espresso.idling.CountingIdlingResource

object EspressoIdlingResource {

    private const val RESOURCE = "GLOBAL"
    private val idlingResource = CountingIdlingResource(RESOURCE)

    fun increment() = idlingResource.increment()

    fun decrement() = idlingResource.decrement()

}